### PR TITLE
ac-66 get people <category> api

### PR DESCRIPTION
**Pull request by ExtraGravity**
Removed the get all people api for now.
For times sake, if we need to get everyone, we can just get each type
individually.
Right now only certain attribute types are hard coded, but not people.
So if we add an new people category that doesn't use any new attributes,
code doesn't need to change.

Also, I'm pretty sure there's some functional magic that I can do to make this shorter, but I don't know what it is, so please enlighten me :)

### CODE REVIEW

File: src/database/people/index.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
+import dbConnection, { tableNames, execute } from '../connection';
+
+const R = require('ramda');
+
+const peopleTable = () => dbConnection()(tableNames.PEOPLE);
+const peopleAttributesTable = () => dbConnection()(tableNames.PEOPLE_ATTRIBUTES);
+
+/**
+ * Represents a product transaction, except for it's typeid
```
Comment (beaesguerra) [inline src/database/people/index.ts]: need to update the comment :)

File: src/database/people/index.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  lastName: string;
+  phoneNumber: string;
+  phoneArea: string;
+  phoneCountry: string;
+  companyName: string;
+  lastModified: string;
+
+  username?: string;
+  paymentFrequency?: string;
+  notes?: string;
```
Comment (beaesguerra) [inline src/database/people/index.ts:51]: should we be returning the type the person is? or nah?

File: src/database/people/index.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+}
+
+interface PeopleAttribute {
```
Comment (beaesguerra) [inline src/database/people/index.ts]: should this say db after so it's consistent?

File: src/database/people/index.ts [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+  personuuid: string;
+  attrname: string;
+  attrvalue: string;
+}
+
+const builders = {
+  /** Get all product transactions of a certain type */
```
Comment (beaesguerra) [inline src/database/people/index.ts]: update comment

File: src/database/people/index.ts [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+  getPeople(peoplecategoryname: string) {
+    return peopleTable().select('*')
+    .join(tableNames.PEOPLE_CATEGORIES,
+      tableNames.PEOPLE + '.peoplecategoryid',
+      tableNames.PEOPLE_CATEGORIES + '.peoplecategoryid')
+    .where({ peoplecategoryname });
```
Comment (beaesguerra) [inline src/database/people/index.ts:61]: ohh hey this is what i was trying to work on before lol :P

File: src/database/people/index.ts [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+  },
+
+  /** get all attributes and their values for a products certain type */
```
Comment (beaesguerra) [inline src/database/people/index.ts]: update comments :)

File: src/database/people/index.ts [hunk 7]
```diff
@@ -0,0 +1,115 @@
+import dbConnection, { tableNames, execute } from '../connection';
+
+const R = require('ramda');
```
Comment (beaesguerra) [inline src/database/people/index.ts]: I don't think we are supposed to be importing like this. 
Should be like this: `import * as R from 'ramda';`

File: src/database/people/index.ts [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+};
+
+/** Get all people of a certain category */
+export async function getPeople(personCategory: string): Promise<Person[]> {
+  const people = await execute<PersonDb[]>(builders.getPeople(personCategory));
+  const attrValues = await execute<PeopleAttribute[]>(builders.getPeopleAttributeValues(personCategory));
+  const results : Person[] = [];
+
+  people.forEach(function (item) {
+    const isMatchingUuid = R.propEq('personuuid', item.personuuid);
```
Comment (beaesguerra) [inline src/database/people/index.ts]: hmm it might just be me but I find this syntax difficult to understand. This is just making a function right?

Comment (beaesguerra) [reply src/database/people/index.ts]: what's the benefit of using ramda over normal functions?

File: src/routers/people/index.ts [hunk 9]
```diff
@@ -1,166 +1,133 @@
-import createRouter from '@/utilities/functions/createRouter';
```
Comment (beaesguerra) [inline src/routers/people/index.ts:1]: why did this file change? looks the same?

**Review by beaesguerra — CHANGES_REQUESTED** [review]

File: src/routers/people/index.ts (hunk 9 above)
Comment (ExtraGravity) [reply src/routers/people/index.ts:1]: line endings :)

File: src/database/people/index.ts (hunk 7 above)
Comment (ExtraGravity) [reply src/database/people/index.ts]: no problem, but what's the difference?

File: src/database/people/index.ts (hunk 8 above)
Comment (ExtraGravity) [reply src/database/people/index.ts]: I find this easier to read, this functional approach imo is cleaner than having more loops.

File: src/database/people/index.ts (hunk 7 above)
Comment (beaesguerra) [reply src/database/people/index.ts]: I'm not sure but Brad says so :P

**Review by beaesguerra — APPROVED** [review]

**Pull request merged**